src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs

```diff
@@ -187,81 +174,107 @@ await using (var ms = new MemoryStream())
+                    await ms.CopyToAsync(ResponseBodyStream());
                 }
 
                 return null;
             }, operationId, token: token);
         }
 
-        private static async Task WriteLogFile(ZipArchive archive, CancellationToken token = default)
+        private static async Task WriteLogFile(ZipArchive archive, CancellationToken token)
         {
-            token.ThrowIfCancellationRequested();
-
-            var prefix = $"{_serverWidePrefix}/{DateTime.UtcNow:yyyy-MM-dd H:mm:ss}.txt";
-
-            try
+            var prefix = $"{_serverWidePrefix}/{DateTime.UtcNow:yyyy-MM-dd H:mm:ss}.log";
+            var entry = archive.CreateEntry(prefix, CompressionLevel.Optimal);
+            entry.ExternalAttributes = (int)(FilePermissions.S_IRUSR | FilePermissions.S_IWUSR) << 16;
+            await using (var entryStream = entry.Open())
+            await using (var sw = new StreamWriter(entryStream))
             {
-                var entry = archive.CreateEntry(prefix, CompressionLevel.Optimal);
-                entry.ExternalAttributes = ((int)(FilePermissions.S_IRUSR | FilePermissions.S_IWUSR)) << 16;
-
-                await using (var entryStream = entry.Open())
+                try
                 {
-                    LoggingSource.Instance.AttachPipeSink(entryStream);
+                    token.ThrowIfCancellationRequested();
 
+                    LoggingSource.Instance.AttachPipeSink(entryStream);
+                    
                     await Task.Delay(15000, token);
+                }
+                catch (Exception e)
+                {
+                    await sw.WriteAsync($"{DateTime.UtcNow.Add(new TimeSpan(LoggingSource.LocalToUtcOffsetInTicks)):yyyy-MM-ddTHH:mm:ss.fffffffZ}, {e.Message}");
+                }
+                finally
+                {
                     LoggingSource.Instance.DetachPipeSink();
-
-                    await entryStream.FlushAsync(token);
+                    await entryStream.FlushAsync();
                 }
             }
-            catch (OperationCanceledException)
+        }
+
+        private async Task<Stream> GetDebugInfoFromNodeAsync(JsonOperationContext context, RequestExecutor requestExecutor, long operationId, OperationCancelToken token,
+            int timeoutInSec)
+        {
+            var rawStreamCommand = new GetRawStreamResultCommand($"/admin/debug/info-package?operationId={operationId}");
+            var requestExecutionTask = requestExecutor.ExecuteAsync(rawStreamCommand, context);
+
+            var delayTask = Task.Delay(TimeSpan.FromSeconds(timeoutInSec), token.Token);
+
+            try
             {
-                throw;
+                var result = await Task.WhenAny(requestExecutionTask, delayTask);
+                if (result == delayTask)
+                {
+                    await KillOperation();
+                }
+                else
+                {
+                    token.Cancel();
```

Comment:
ping me about this tomorrow

Reply:
You're right. We shouldn't cancel it here.